### PR TITLE
feat(communities): Adds mute category interval

### PR DESCRIPTION
**Pull request by borismelnik**
Part of: #9369

[STATUSGO PR](https://github.com/status-im/status-go/pull/3534)

### What does the PR do

Adds mute interval for categories

### Affected areas

Muting category

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Снимок экрана 2023-05-30 в 20 47 47

### CODE REVIEW

**Review by jrainville — COMMENTED** [review]
Looks good.
Just small comments.

File: src/backend/communities.nim [hunk 1]
```diff
@@ -9,13 +9,23 @@ export response_type
...
+  ])
 
 proc unmuteCategory*(communityId: string, categoryId: string): RpcResponse[JsonNode] {.raises: [Exception].} =
-  let payload = %* [communityId, categoryId]
-  result = callPrivateRPC("unmuteCommunityCategory".prefix, payload)
+  result = callPrivateRPC("unmuteCommunityCategory".prefix, %* [
+    {
+      "communityId": communityId,
+      "categoryId": categoryId,
+      "mutedType": 0, # by default it's 0
```
[the remark this answers is not in the document]
Comment (borismelnik) [reply src/backend/communities.nim:26]: I thought about it... will change the api

**Review by jrainville — APPROVED** [review]

**Comment by borismelnik** [comment]
@jrainville I request your review cos I add some changes to process muting changes from status-go. I tested it and now it works for chats, channels and categories as well. 

As I remember, you asked me about some changes ` SIGNAL_CHAT_MUTED` event, but it used in several places. Am I right or missed something in your request?

**Comment by jrainville** [comment]
@borismelnik ah right I remember. You did not address it here.

Basically, it's quite simple, in the `response` from `muteChat` and `unmuteChat`, there will probably be a `chat` object that represents the updated chat. You can just call `updateOrAddChat` from the chat service with the chat, or even simpler, pass the response to `parseChatResponse`.

That way, we are sure that our chat cache in the service stays up to date with the chat from status-go.

That's part of what Mishka talked about last Friday. When we get a response from status-go, we need to use it and update our state from that one instead of trying to manually update.

There is also a chance that the manual functions and events that we use like `SIGNAL_CHAT_MUTED` will be no longer needed after that.

**Comment by borismelnik** [comment]
> @borismelnik ah right I remember. You did not address it here.
> 
> Basically, it's quite simple, in the `response` from `muteChat` and `unmuteChat`, there will probably be a `chat` object that represents the updated chat. You can just call `updateOrAddChat` from the chat service with the chat, or even simpler, pass the response to `parseChatResponse`.
> 
> That way, we are sure that our chat cache in the service stays up to date with the chat from status-go.
> 
> That's part of what Mishka talked about last Friday. When we get a response from status-go, we need to use it and update our state from that one instead of trying to manually update.
> 
> There is also a chance that the manual functions and events that we use like `SIGNAL_CHAT_MUTED` will be no longer needed after that.

oh I got your point :) will do

File: src/backend/communities.nim [hunk 2]
```diff
@@ -9,13 +9,17 @@ export response_type
...
-  let payload = %* [communityId, categoryId]
-  result = callPrivateRPC("muteCommunityCategory".prefix, payload)
+proc muteCategory*(communityId: string, categoryId: string, interval: int): RpcResponse[JsonNode] {.raises: [Exception].} =
+  result = callPrivateRPC("muteCommunityCategory".prefix, %* [
+    {
+      "communityId": communityId,
+      "categoryId": categoryId,
+      "mutedType": interval,
+    }
+  ])
```
Comment (MishkaRogachev) [inline src/backend/communities.nim:19]: extra space

**Review by MishkaRogachev — APPROVED** [review]
Good!

**Comment by borismelnik** [comment]
@jrainville unfortunately, no chats in response. I can add it, but in separate PR

**Comment by jrainville** [comment]
> @jrainville unfortunately, no chats in response. I can add it, but in separate PR

Ok no worries, I'll open an issue so we don't forget and we'll fix later

**Review by jrainville — APPROVED** [review]

**Pull request merged**